src/test/java/net/sf/jabref/model/database/KeyChangeListenerTest.java

```diff
@@ -0,0 +1,98 @@
+package net.sf.jabref.model.database;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fileformat.BibtexParser;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class KeyChangeListenerTest {
+
+    private BibDatabase db;
+
+
+    @Before
+    public void setUp() throws IOException {
+        Globals.prefs = JabRefPreferences.getInstance();
+        try (FileInputStream stream = new FileInputStream("src/test/resources/testbib/keychangelistener.bib");
+                InputStreamReader fr = new InputStreamReader(stream, StandardCharsets.UTF_8)) {
+            ParserResult result = BibtexParser.parse(fr);
+
+            db = result.getDatabase();
```

Comment:
Please create the database by hand instead of reading it from the file. It also has the advantage that you can save the reference to the entries in private fields.